index.js

```diff
@@ -191,3 +204,11 @@ function zOrder(x, y) {
 
     return x | (y << 1);
 }
+
+function hasAllKeys(tags, requiredKeys) {
+    var tagKeys = tags.filter(function (_, i) { return i % 2 === 0; });
+    for (var i = 0, j = requiredKeys.length; i < j; i++) {
+        if (tagKeys.indexOf(requiredKeys[i]) === -1) return false;
+    }
+    return true;
+}
```

Comment:
Just read this msg after implementing the lookup fix :|
Passing a static integer for numRequiredKeys should improve my last commit. Will add this. Thanks!